android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,629 @@
+				int error = finalizeEvent.getError();
+				if (error != VideoRecordEvent.Finalize.ERROR_NONE) {
+					// error
+				}
+				Uri uri = finalizeEvent.getOutputResults().getOutputUri();
+
+				if (uri.toString().startsWith("content://")) {
+					String[] columns = {
+						MediaStore.Video.VideoColumns.TITLE,
+						MediaStore.Video.VideoColumns.DATA
+					};
+					String[] selectionArgs = { mediaTitle };
+					createFile(
+						MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+						columns,
+						MediaStore.Video.VideoColumns.TITLE + " = ? ",
+						selectionArgs
+					);
+				} else {
+					// normal file
+					File file = new File(uri.getPath());
+					TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(file.getPath(), false));
+					KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+					successCallback.callAsync(callbackContext, response);
+				}
+
+				if (cameraActivity != null && autohide) {
+					if (recording != null) {
+						recording.close();
+						recording = null;
+					}
+					cameraActivity.finish();
+				}
+			}
+			// All events, including VideoRecordEvent.Status, contain RecordingStats.
+			// This can be used to update the UI or track the recording duration.
+			RecordingStats recordingStats = videoRecordEvent.getRecordingStats();
+			long recordingTime = (recordingStats.getRecordedDurationNanos() / 1000 / 1000);
+			if (isRecording && videoMaximumDuration > 0 && recordingTime >= videoMaximumDuration) {
+				stopVideoCapture();
+			}
+			if (isRecording && recordingCallback != null) {
+				recordingDict.put("duration", recordingTime);
+				recordingDict.put("size", recordingStats.getNumBytesRecorded());
+				recordingCallback.callAsync(callbackContext, recordingDict);
+			}
+		};
+
+		if (cameraActivity.hasAudioRecorderPermissions()) {
+			recording = pendingRecording.withAudioEnabled().start(executor, vre);
+			isRecording = true;
+		} else {
+			recording = pendingRecording.start(executor, vre);
+			isRecording = true;
+		}
+	}
+
+	public static void pauseVideoCapture()
+	{
+		if (recording != null) recording.pause();
+		isRecording = false;
+	}
+
+	public static void resumeVideoCapture()
+	{
+		if (recording != null) recording.resume();
+		isRecording = true;
+	}
+
+	public static void stopVideoCapture()
+	{
+		if (recording != null) recording.stop();
+		isRecording = false;
+	}
+
+	public static void setFlashMode(int value)
+	{
+		cameraFlashMode = value;
+		if (imageCapture != null) {
+			imageCapture.setFlashMode(cameraFlashMode);
+		}
+	}
+
+	public static void hide()
+	{
+		if (recording != null) {
+			recording.close();
+			recording = null;
+		}
+		if (cameraActivity != null) {
+			cameraActivity.finish();
+		}
+	}
+
+	private static String createExternalMediaName()
+	{
+		TiApplication app = TiApplication.getInstance();
+		String normalizedAppName = app.getAppInfo().getName();
+		normalizedAppName = normalizedAppName.replaceAll("[^\\w.-]", "_");
```

Comment:
Do we use this one in our CLI as well? Just to be sure.

Reply:
I took that from https://github.com/tidev/titanium_mobile/blob/3362dd2154714c817185096708c7a4b7e4b5bf6f/android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java#L859 to create a custom filename that is without any special characters